rocky/katalogus/forms/plugin_settings.py

```diff
@@ -12,65 +15,41 @@ class PluginSchemaForm(forms.Form):
         "required": _("This field is required."),
     }
 
-    def __init__(self, plugin_schema, *args, **kwargs):
+    def __init__(self, plugin_schema: Dict, values: Dict, *args, **kwargs):
         super().__init__(*args, **kwargs)
         self.plugin_schema = plugin_schema
+        self.values = values
         self.populate_fields()
 
     def populate_fields(self):
-        fields = self.plugin_schema["properties"]
-        required_fields = self.plugin_schema["required"]
-        help_text = ""
-        for field_name, field_props in fields.items():
-            field_type = FIELD_TYPES[field_props["type"]]
-            if "description" in field_props:
-                help_text = field_props["description"]
+        for field_name, field_props in self.plugin_schema["properties"].items():
             kwargs = {
-                "required": field_name in required_fields,
+                "required": field_name in self.plugin_schema["required"],
                 "label": field_props.get("title", field_name),
-                "help_text": _(help_text),
+                "help_text": _(field_props.get("description", "")),
                 "error_messages": self.error_messages,
             }
             if field_props["type"] == "string":
                 kwargs["max_length"] = min(
                     MAX_SETTINGS_VALUE_LENGTH, field_props.get("maxLength", MAX_SETTINGS_VALUE_LENGTH)
                 )
-            self.fields[field_name] = field_type(**kwargs)
 
+            if field_name in self.values:
+                kwargs["initial"] = self.values[field_name]
 
-class PluginSettingAddEditForm(forms.Form):
-    """Form for adding a single setting or multiple settings, use setting name
-    to populate schema field propertis into form field."""
+            field_type = FIELD_TYPES[field_props["type"]]
+            self.fields[field_name] = field_type(**kwargs)
 
-    error_messages = {
-        "required": _("This field is required."),
-    }
+    def clean(self):
+        cleaned_data = super().clean()
 
-    def __init__(self, plugin_schema, setting_name, setting_value=None, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.plugin_schema = plugin_schema
-        self.setting_name = setting_name
-        self.setting_value = setting_value
-        self.populate_field()
+        # The form assigns "" and in some scenario's null to all unfilled (optional) fields
+        cleaned_data = {key: value for key, value in cleaned_data.items() if value is not None and value != ""}  # noqa
```

Comment:
What error does this violate and why should we suppress this with a `noqa`?

Reply:
What is?

Reply:
It would be better to use `# noqa: PLC1901`. That makes it clear what rule is silenced and doesn't silence any other rule.

Reply:
> Because we should accept the value being 0 for instance, so something like `if value` or `if bool(value)` will not work. Checking both the type _and_ truthiness is overkill

I was under the impression that cleaned data is always a dictionary of string-(optional)string pairs, that's why I was surprised that value could be anything other than a `str`.